From 62cb8d22b7ed01882182ab40be88c68d6b7704b1 Mon Sep 17 00:00:00 2001
From: Mohammed Rafi KC <rkavunga@redhat.com>
Date: Tue, 26 Apr 2016 14:52:29 +0530
Subject: [PATCH 098/104] gfapi: set need_lookup flag on response list

     back port of : http://review.gluster.org/#/c/14098/

We set need_lookup flag for entries returned by readdirp
to force lookup. Currently we are setting on the previously
stored list, rather than response list returned by readdirp.

This patch will iterate over current list returned by readdirp
and will set need_lookup flag.

Back port of>
>Change-Id: Ibd6fcbc188f4c87f40ece7a9dcda27645401c240
>BUG: 1330476
>Signed-off-by: Mohammed Rafi KC <rkavunga@redhat.com>
>Reviewed-on: http://review.gluster.org/14073
>NetBSD-regression: NetBSD Build System <jenkins@build.gluster.org>
>Reviewed-by: Niels de Vos <ndevos@redhat.com>
>Reviewed-by: Rajesh Joseph <rjoseph@redhat.com>
>Smoke: Gluster Build System <jenkins@build.gluster.com>
>CentOS-regression: Gluster Build System <jenkins@build.gluster.com>

>(cherry picked from commit 537557da59876536d33cd25a8ef485e7b5fe8849)

Change-Id: I9efe9de15afdd4099b9db870d6c815361789faff
BUG: 1322247
Signed-off-by: Mohammed Rafi KC <rkavunga@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/73264
Reviewed-by: Rajesh Joseph <rjoseph@redhat.com>
Tested-by: Rajesh Joseph <rjoseph@redhat.com>
---
 api/src/glfs-fops.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/api/src/glfs-fops.c b/api/src/glfs-fops.c
index fdffc7b..98be987 100644
--- a/api/src/glfs-fops.c
+++ b/api/src/glfs-fops.c
@@ -2443,7 +2443,7 @@ glfd_entry_refresh (struct glfs_fd *glfd, int plus)
                          * to a race where a fop comes after inode link
                          * but before setting need_lookup flag.
                          */
-                        list_for_each_entry (entry, &glfd->entries, list) {
+                        list_for_each_entry (entry, &entries.list, list) {
                                 if (entry->inode)
                                         inode_set_need_lookup (entry->inode, THIS);
                         }
-- 
1.7.1

